apps/performance/christmas-web-worker/src/app/app.component.ts

```diff
@@ -10,23 +17,62 @@ import { UnknownPersonComponent } from './unknown-person/unknown-person.componen
   selector: 'app-root',
   template: `
     <unknown-person [step]="loadingPercentage()" class="relative grow" />
-    <button
-      class="my-3 w-fit self-center rounded-md border border-white px-4 py-2 text-2xl text-white"
-      (click)="discover()">
-      Discover
-    </button>
+
+    @if (loadingPercentage() === 0) {
+      <button
+        class="my-3 w-fit self-center rounded-md border border-white px-4 py-2 text-2xl text-white"
+        (click)="discover()">
+        Discover
+      </button>
+    }
+
     <div class="p-1 text-white">Progress: {{ loadingPercentage() }}%</div>
   `,
   host: {
     class: `flex flex-col h-screen w-screen bg-[#1f75c0]`,
   },
 })
 export class AppComponent {
-  private heavyCalculationService = inject(HeavyCalculationService);
+  private readonly heavyCalculationService = inject(HeavyCalculationService);
+
+  private readonly webWorkerLoadingPercentage: WritableSignal<number> =
```

Comment:
you don't have to type your property, TS will infer it for you. easier to read and TS is always right. 

You can type it if you want to narrow it.

Reply:
we have a convention at work to type all properties, so it it probably a habit :) 
fixed

Reply:
Yes many companies have this habits, but I don't really get it. It add more code, and TS is more accurate than we are. 😅

Reply:
Well as far as i remember the main points were:
1 when you make the review, you can see types in the code, so you have more info about the code and better understanding of it.
2 when you write code you should explicitly specify types to make sure that functions return exactly what you expect them to return